dataset/to_unit.cpp

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Jan-Lukas Wynen
+#include <units/units.hpp>
+
+#include "scipp/dataset/to_unit.h"
+#include "scipp/variable/to_unit.h"
+
+namespace scipp::dataset {
+
+DataArray to_unit(const DataArray &array, const units::Unit &unit,
+                  const CopyPolicy copy) {
+  return DataArray(to_unit(array.data(), unit, copy), array.coords(),
+                   array.masks(), array.attrs(), array.name());
```

Comment:
Masks should also be affected by the copy policy, compare other operations that modify data, e.g., https://github.com/scipp/scipp/blob/94bb70486d9a9a7296393f9fab0089ec771d62d4/templates/dataset_unary.cpp.in#L10-L12

Reply:
Would this depend on whether the data is copied? Or would it always not copy if `copy == CopyPolicy::TryAvoid`?

Reply:
I think it should depend on whether data is copied or not.